homeassistant/components/sensorpush_cloud/sensor.py

```diff
@@ -0,0 +1,168 @@
+        entity_registry_enabled_default=False,
+        translation_key="battery_voltage",
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_DEWPOINT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        entity_registry_enabled_default=False,
+        translation_key="dewpoint",
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_HUMIDITY,
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_SIGNAL_STRENGTH,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_TEMPERATURE,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_VAPOR_PRESSURE,
+        device_class=SensorDeviceClass.PRESSURE,
+        entity_registry_enabled_default=False,
+        translation_key="vapor_pressure",
+        native_unit_of_measurement=UnitOfPressure.KPA,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SensorPushCloudConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SensorPush Cloud sensors."""
+    coordinator: SensorPushCloudCoordinator = entry.runtime_data
+    device_ids = await coordinator.async_get_device_ids()
+    async_add_entities(
+        SensorPushCloudSensor(coordinator, entity_description, device_id)
+        for entity_description in SENSORS
+        for device_id in device_ids
+    )
+
+
+class SensorPushCloudSensor(
+    CoordinatorEntity[SensorPushCloudCoordinator], SensorEntity
+):
+    """SensorPush Cloud sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: SensorPushCloudCoordinator,
+        entity_description: SensorEntityDescription,
+        device_id: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self.device_id = device_id
+
+        if device_id not in self.coordinator.devices:
+            LOGGER.warning(f"Ignoring inactive device: {device_id}")
+            self._attr_available = False
+            return
+
+        device = self.coordinator.devices[device_id]
+        self._attr_unique_id = f"{device[ATTR_DEVICE_ID]}_{entity_description.key}"
```

Comment:
what's the difference between `device_id` and `device[ATTR_DEVICE_ID]`?

Reply:
These should be the same value. `device_id` is returned from the API as a key, however it is also embedded in the response data. I kept the calling conventions similar to other accesses for consistency.